android-project/app/src/main/java/oceantreasur/es/fragments/AnimationController.java

```diff
@@ -0,0 +1,135 @@
+import static oceantreasur.es.view.AnimationConstants.FISH_MARGIN_BOTTOM;
+import static oceantreasur.es.view.AnimationConstants.FISH_MARGIN_TOP;
+import static oceantreasur.es.view.AnimationConstants.MAX_BIG_FISHES;
+import static oceantreasur.es.view.AnimationConstants.MAX_DURATION;
+import static oceantreasur.es.view.AnimationConstants.MAX_SMALL_FISH;
+import static oceantreasur.es.view.AnimationConstants.MAX_SPAWNING_POINT_IN_PERCENTS;
+import static oceantreasur.es.view.AnimationConstants.MAX_TIME_OFFSET_BIG_FISH;
+import static oceantreasur.es.view.AnimationConstants.MAX_TIME_OFFSET_SMALL_FISH;
+import static oceantreasur.es.view.AnimationConstants.MIN_DURATION;
+import static oceantreasur.es.view.AnimationConstants.MIN_SPAWNING_POINT_IN_PERCENTS;
+import static oceantreasur.es.view.AnimationConstants.MIN_TIME_OFFSET_BIG_FISH;
+import static oceantreasur.es.view.AnimationConstants.MIN_TIME_OFFSET_SMALL_FISH;
+import static oceantreasur.es.view.AnimationConstants.SPAWNING_POINT_OF_BIG_FISH;
+import static oceantreasur.es.view.ScreenUtils.getDimensionOfFishInPx;
+import static oceantreasur.es.view.ScreenUtils.getScreenHeight;
+import static oceantreasur.es.view.ScreenUtils.getScreenWidth;
+
+
+public class AnimationController {
+
+    private ArrayList<View> smallFishViews = new ArrayList<>();
+    private ArrayList<View> bigFishViews = new ArrayList<>();
+    private int smallFishBeginOffset = getDimensionOfFishInPx(small_fish_width);
+    private int bigFishBeginOffset = getDimensionOfFishInPx(R.dimen.big_fish_width);
+
+    AnimationController(RelativeLayout background) {
+        inflateFishViews(background);
+        moveAnimation();
+    }
+
+    private void inflateFishViews (RelativeLayout background) {
+
+        LayoutInflater Li = LayoutInflater.from(OceanTreasuresApplication.getStaticContext());
+
+        for(int i = 0; i < MAX_SMALL_FISH; i++) {
+            smallFishViews.add(Li.inflate(small_fish, null));
+            background.addView(smallFishViews.get(i));
+        }
+
+        for(int i = 0; i < MAX_BIG_FISHES; i++) {
+            bigFishViews.add(Li.inflate(big_fish, null));
+            background.addView(bigFishViews.get(i));
+        }
+    }
+
+    private void moveAnimation() {
+        setupAnimation(smallFishViews, smallFishBeginOffset);
+        setupAnimation(bigFishViews, bigFishBeginOffset);
+    }
+
+    private void setupAnimation(ArrayList<View> views, int offset) {
+        for(int i = 0; i < views.size(); i++) {
+            startAnimation(views.get(i), generateRandomIntegerInRange(MIN_DURATION, MAX_DURATION), offset);
+        }
+    }
+
+    private int choseRandomYPosition() {
+        int height = getScreenHeight();
+
+        int yPosition = generateRandomIntegerInRange(1, height);
+
+        if(yPosition > height * MAX_SPAWNING_POINT_IN_PERCENTS) {
+            yPosition -= (getDimensionOfFishInPx(small_fish_height) + FISH_MARGIN_BOTTOM);
```

Comment:
Don't use the constant for the height for the small fish. Pass the fish height to this method and work with the actual height of the current fish.